grammar-gen/src/codegen/rt.rs

```diff
@@ -498,6 +524,33 @@ pub(crate) fn runtime_base(entry_points: impl IntoIterator<Item = Ident>) -> Tok
 
         impl<Span> Eq for TokenBuffer<Span> {}
 
+        impl<Span> Ord for TokenBuffer<Span>
+        where
+            Span: Copy + 'static
+        {
+            fn cmp(&self, other: &TokenBuffer<Span>) -> Ordering {
+                self
+                    .len()
+                    .cmp(&other.len())
+                    .then_with(||
+                        self
+                            .as_slice()
```

Comment:
Maybe create a `tokens` method that returns an `impl Iterator<Item = TokenDescription>`.